src/java.desktop/macosx/classes/sun/awt/CGraphicsEnvironment.java

```diff
@@ -139,13 +143,19 @@ void _displayReconfiguration(int displayId, boolean removed) {
         rebuildDevices();
     }
 
-    @Override
-    @SuppressWarnings("removal")
-    protected void finalize() throws Throwable {
-        try {
-            super.finalize();
-        } finally {
-            deregisterDisplayReconfiguration(displayReconfigContext);
+    private static class CGEDisposerRecord implements DisposerRecord {
+        private long displayReconfigContext;
+
+        CGEDisposerRecord(long ptr) {
+            displayReconfigContext = ptr;
+        }
+
+        public void dispose() {
+            try {
+                deregisterDisplayReconfiguration(displayReconfigContext);
+            } catch (Throwable t) {
+                // Runs on disposer thread, can't allow an exception to escape.
```

Comment:
What do you think about moving this "catch Throwable" into Disposer itself? As of now since https://github.com/openjdk/jdk/pull/9890 we ignore "Exception" only.

Reply:
I can do that. I can't think of a problem with it.
I should probably then do the same for the other catches of Exception in that file.

I'm a bit on the fence about whether I should
1. keep the print statements there
2. remove them
3. hide them behind a debugging option

I think I'll leave them alone and if we need to we can adjust in another PR.